ChatSX/components/Poll/index.tsx

```diff
@@ -0,0 +1,1689 @@
+              },
+            });
+          }
+          break;
+        }
+
+        case 1: {
+          if (
+            selectedPolls.length <= item.multiple_select_no &&
+            selectedPolls.length > 0
+          ) {
+            // show submit poll button
+            setShouldShowSubmitPollButton(true);
+          } else {
+            // hide submit poll button
+            setShouldShowSubmitPollButton(false);
+          }
+          break;
+        }
+
+        case 2: {
+          if (selectedPolls.length >= item.multiple_select_no) {
+            // show submit poll button
+            setShouldShowSubmitPollButton(true);
+          } else {
+            // hide submit poll button
+            setShouldShowSubmitPollButton(false);
+          }
+          break;
+        }
+
+        default: {
+          // hide submit poll button
+          setShouldShowSubmitPollButton(false);
+        }
+      }
+    }
+  };
+
+  // this useEffects monitors the interaction in poll options, according to which we have to show submit button
+  useEffect(() => {
+    submitPollLogics();
+  }, [selectedPolls]);
+
+  // check if user already answered on th poll or not
+  useEffect(() => {
+    let count = 0;
+    const res = pollsArr?.forEach((poll: any) => {
+      if (poll.is_selected === true) {
+        count = count + 1;
+      }
+    });
+    if (count > 0) {
+      setAllowAddOption(false);
+      setShouldShowVotes(true);
+
+      // deffered poll show edit button state updation logic
+      if (item?.poll_type === 1) {
+        setShowResultsButton(true);
+      }
+    } else {
+      setAllowAddOption(item?.allow_add_option);
+      setShouldShowVotes(false);
+    }
+
+    setPollVoteCount(count);
+  }, [pollsArr]);
+
+  // Poll end timer logic
+  useEffect(() => {
+    const difference = item?.expiry_time - Date.now();
+
+    if (difference > 0) {
+      setHasPollEnded(false);
+    } else {
+      setHasPollEnded(true);
+    }
+    if (difference > 0) {
+      let timer = setTimeout(() => {
+        setHasPollEnded(true);
+      }, difference);
+
+      return () => {
+        clearTimeout(timer);
+      };
+    }
+  }, []);
+
+  // this function triggers when have to hide anonymous poll modal
+  const hideAnonymousPollModal = () => {
+    setIsAnonymousPollModalVisible(false);
+  };
+  // this function resets showResult state
+  const resetShowResult = () => {
+    let arr = pollsArr.map((item: any) => {
+      return {
+        ...item,
+        is_selected: false,
+        percentage: 0,
+        no_votes: 0,
+      };
+    });
+    setPollsArr([...arr]);
+    setShowResultsButton(false);
+    setShouldShowVotes(false);
+    setSelectedPolls([]);
+  };
+
+  useEffect(() => {
+    setPollsArr(item?.polls);
+  }, [item?.polls]);
+
+  // API to reload the existing poll conversation
+  async function reloadConversation() {
```

Comment:
why we are making API call in UI class, by this way, will not be able to segregate UI and Business logic